src/utils/input_parser.cpp

```diff
@@ -51,19 +51,18 @@ inline double get_double(const rapidjson::Value& object, const char* key) {
 
 inline Amount get_amount(const rapidjson::Value& object,
                          const char* key,
-                         unsigned size) {
-  // Default to zero amount with provided size.
-  Amount amount(size);
+                         const Amount& zero_amount) {
+  Amount amount(zero_amount);
```

Comment:
This init is probably simpler then, but I don't think this part of (internal) code can or even should be made non-aware of amount dimensions, since we have to check dimension consistency right below.